### DISCUSSION TITLE
Any chance of larger displays?

### DISCUSSION THREAD

**Discussion by deadesq** [outdoorbits/little-backup-box#271] [Ideas]
Hi--I'm loving using LBB on my Pi4, and it works great with my phone to watch the process.  However, it would be great to have a fully-integrated solution if that's on the roadmap anywhere.  I have a 3x2 inch LCD (approx) display on my Pi4, and while I know that small displays are supported, when I install the drivers for the display, LBB no longer functions, and when the Pi4 boots up, it just goes to the command line.  Disclaimer--I'm a photographer, not a programmer.  Well, I used to program, but that was back when you had to do an IBM punch card for each line of code.  So, please be gentle with the comments... TIA

**Comment by outdoorbits** [comment]
Hi, I would expect your display is connected to a video output, not to 
GPIO like the displays used for lbb? This would mean, you need a 
graphical system, not the lite version of the os - and this would 
interact with the storage management of lbb and cause new problems. I 
don't know what your drivers are doing and can't explain, why they stop 
the booting process of lbb.

But, most important, this is not the way lbb uses the display. Why don't 
you just print the case 
https://github.com/outdoorbits/case-for-little-backup-box and assemble 
the (cheap) components?

**Reply by deadesq** [reply]
Thanks for the quick reply.  I'm working through various workarounds, but I found something that might work.  To answer your question, I don't have a 3D printer, so that's something I can't take on right now.  What I did is install a graphical desktop environment, which boots fine.  I've attached an image of the setup, I hope.  

The video is connected to the GPIO directly.  What I'm just hoping to do is have the status of LBB show up on the video, rather than on a separate device like my phone.  I know it does this for the displays that work with LBB, but I can't figure out a way to have that show up on the video display I have installed.   

![IMG_1482](https://github.com/outdoorbits/little-backup-box/assets/153684931/c3288e5d-f976-4e31-a8e7-6ecfe7b5f241)

**Comment by outdoorbits** [comment]
If you can manage to...

  - auto login to the graphical UI
  - autostart a web browser to open http://localhost
  - inhibit auto mount of your graphical surface

...you did it! If you find a way to do so, I would like to add this option.

**Reply by deadesq** [reply]
Okay--I'm going to work on that over the next day or two.  Preliminary indications is that it will work.  I did run a successful backup so it is still working in the background, but I want to see if I can monitor it in real time with the built-in web browser.  No reason that shouldn't work.  I'll report back.

**Comment by deadesq** [comment]
Okay!  Success of a sort.  Rough workflow--I installed the full graphic UI, installed the drivers for the 3.5" LCD, then installed LBB.  A few minor tweaks of the LCD that are not relevant here, but I can share if desired.  I basically needed it to show more of the Chromium web page...  Anyway, I'm able to open https://localhost, and the LBB GUI interface comes up.  I scrolled to the log window and watched a backup in real time.  The backup worked as it should have, and when it was done, I was able to view the photos on the Raspberry Pi directly, rather than using my phone/tablet.  It's a fully functional setup as I was hoping to do!  There are some tweaks, though, that I will work on.  First, I'd like to have a kiosk mode for Chromium at startup with the LBB interface.  I think I can eventually figure that out.  Second, I would really like to not have to log in to LBB, since I either have to plug in a keyboard, or maybe I can install a touchscreen keyboard.  I'd rather not do it if possible.  But since I first saw a Gnarbox, I've wanted to use a Raspberry Pi for this purpose.  I finally got there tonight!
![IMG_1498 copy](https://github.com/outdoorbits/little-backup-box/assets/153684931/873f929c-8bdc-4855-9d5f-49283c9891b0)

**Comment by outdoorbits** [comment]
This sounds fantastic! I think I could manage to allow access to the web UI without password for localhost.
How did you manage conflicts of auto mount?
Is there a way to automatically start the web browser after boot? Maybe directly in full screen mode?

**Comment by deadesq** [comment]
Not sure there were any auto-mount conflicts, because it just seemed to work.  I did figure out how to get the web browser into kiosk mode at boot, though.  I added @chromium-browser --kiosk http://localhost/ to the autostart file at /home/pi/.config/lxsession/LXDE-pi/autostart.  Voila.  Now, to deal with the pesky login for that page.  It saved the password, so I have to just tap on the screen now, but here's what it looks like.
![IMG_1499](https://github.com/outdoorbits/little-backup-box/assets/153684931/0432a3ee-e0e5-4aab-b5d9-17e52382e01b)

**Comment by deadesq** [comment]
And a little video in action.

https://github.com/outdoorbits/little-backup-box/assets/153684931/b6ebe29e-4901-4179-9d18-abb23d52554a

**Comment by outdoorbits** [comment]
This is really next level! Is it possible to configure all that by a 
bash script

  	* Create a user for linux gui: easy
  	* define auto login without password after boot: have to check
  	* define auto start for the browser connecting to localhost: you did 
it

If you like, we can develop this and include it into the main branch.

**Comment by deadesq** [comment]
There is an auto-login without password after boot.  raspi-config -> system options -> Boot/AutoLogin -> B2 allows login without the GUI, and B4 allows login with the GUI.  I think the simple path without a GUI is to have a script at login that runs Chromium and automatically opens localhost.  I did it with the GUI because I wanted to be able to see what I was doing a little better.  I'm sure the same could be done with a standalone install of Chromium rather than the full desktop GUI.  

I looked around in the php files to see if I could remove the login for the localhost webpage, but I'm not seeing what I'm looking for.  I'd like to comment it out and see if I can load localhost without logging in.

**Reply by deadesq** [reply]
Figured out the localhost login issue.  I just commented out everything in /etc/apache2/includes/password.conf.  Now, it just boots right to LBB as I've been trying to achieve.  

What I may work on next is optimizing the LBB pages for the small display.  But it's working exactly like I'd hoped.  I'll post my progress as I go.

**Comment by outdoorbits** [comment]
I just changed the apache configuration to allow access to the web ui from localhost without password. So you don't lose your protection from external devices. After update just set a new password to bring the new config in place.

**Reply by deadesq** [reply]
I did the update, and it's working as expected.  Thanks!  I really like the setup with the bigger screen and metal case--I look at it, and it really is a little backup box.

**Comment by outdoorbits** [comment]
I just uploaded a new branch "[desktop](https://github.com/outdoorbits/little-backup-box/tree/desktop)". It contains the script setup-graphical-environment.sh witch should arrange everything needed for a desktop usage.
Up to now there are two major problems:

1. Auto mount is not disabled by the script.
2. The browser doesn't start

Do you see solutions?

**Reply by outdoorbits** [reply]
One step ahead: Auto mount is disabled now but I can't get autostart working.

**Comment by outdoorbits** [comment]
For me it's working now. Would you please test? You need a new image, please install your drivers first and after that lbb. If you confirm, I would put this into documentation.

**Reply by deadesq** [reply]
Will do this weekend.  I have a spare card that I will install as new.

**Comment by deadesq** [comment]
It worked perfectly!  Only suggestion for the documentation is that you note you should install the full graphical environment if you want to go this path.  I tried it initially with just the bare image with no desktop environment, thinking you might have included just the bare minimum, but then I re-did it with the GUI.  I successfully backed up my images without a hitch.  Photos and video attached.  And as a bonus, I did it entirely on a battery pack to make sure this will work in the field.  

https://github.com/outdoorbits/little-backup-box/assets/153684931/ba78e734-cabf-4c53-abb8-70658815b6ef
![IMG_1514](https://github.com/outdoorbits/little-backup-box/assets/153684931/24fd28ee-b2e2-47ed-9a4e-8f27991f7aca)
![IMG_1515](https://github.com/outdoorbits/little-backup-box/assets/153684931/0da3da94-34ea-43eb-84bc-f9f5ea4b1d81)

**Comment by outdoorbits** [comment]
That's great news. I'll rework the documentation. If you find the time, could you please take a photo on a neutral background showing the backup screen and the connections? I would like to publish it on the GitHub page. Thank you!

**Reply by deadesq** [reply]
![lbb3](https://github.com/outdoorbits/little-backup-box/assets/153684931/19d60497-1ab2-4736-b715-318a9800d92d)
![lbb2](https://github.com/outdoorbits/little-backup-box/assets/153684931/ab067d47-52a2-4808-84aa-668972f8008e)
![lbb1](https://github.com/outdoorbits/little-backup-box/assets/153684931/35ed8c29-a998-43e4-a26b-dda9d47130f4)

I hope these are what you want.  Also, here's the LCD and case I used.  https://www.amazon.com/Raspberry-Cooling-Heatsink-480x320-Resolution/dp/B0936R564K/

**Comment by deadesq** [comment]
I also see this in the logs when it starts up, but I don't see anywhere where this image gets displayed.  

IMAGE=/var/www/little-backup-box/tmp/ip-qr-link-192-168-12-28_64_online.png

**Reply by outdoorbits** [reply]
It's just for the IP mail and for small oled displays.

**Comment by deadesq** [comment]
A couple things I've noticed in the last day with the latest build...first, it used Firefox as the default.  I wasn't expecting that, so it hung up because it said the page was insecure.  I changed it to chromium which is how I've tested it, and it booted as I am used to seeing.  But, now I can't control the screen at all like before.  I'm not sure what about the kiosk mode changed, but I would need a keyboard to scroll the page now.  I'm sure it's an easy fix, but I haven't figured it out yet.  I'll let you know if I do.

**Comment by outdoorbits** [comment]
I had some issues about chromium so I switched to firefox.Did you change anything in the address http://localhost? The warning about the insecure website shouldn't come up, if you don't use https.

**Comment by deadesq** [comment]
Weird.  I did a reinstall, and the problem went away. I notice you put the little IP address graphic in there at the bottom, too.  Very helpful.Sent from my iPadOn Jan 29, 2024, at 11:11 PM, outdoorbits ***@***.***> wrote:﻿
I had some issues about chromium so I switched to firefox.Did you change anything in the address http://localhost? The warning about the insecure website shouldn't come up, if you don't use https.

**Reply by deadesq** [reply]
Is there a way to try out Chromium instead of Firefox?  I didn't see anything in the autostart file, so I'm assuming it's making the selection somewhere else.

**Comment by outdoorbits** [comment]
Yesterday I added a virtual keyboard. It is only displayed if allowed by configuration (default=true) and on requests from localhost. Maybe you could check this in your device. As you know, there are some text input fields in settings.

**Reply by deadesq** [reply]
I did the update, and the keyboard shows up when I try to enter a new global password, but the keyboard takes up the entire screen, so I can't see the password dialog area.  Any way to make the keyboard small?

**Comment by outdoorbits** [comment]
Unfortunately it's very poor documented and I don't expect that it's possible to resize. I'll have a look but if not, I hope it does it's job anyway.

**Comment by outdoorbits** [comment]
Did the virtual keyboard shrink ab bit on your system? 
I added a wiki page: https://github.com/outdoorbits/little-backup-box/wiki/03a.-Installation-on-a-touchscreen-system

**Reply by deadesq** [reply]
Wiki is great.  I did the update, and this is what the keyboard looks like.  

![IMG_1570](https://github.com/outdoorbits/little-backup-box/assets/153684931/425146d9-a766-4402-80ee-889a9c782f42)

**Comment by deadesq** [comment]
Okay--walkthrough video is live.  https://youtu.be/qAEebY8UlRc

**Comment by outdoorbits** [comment]
Hi,

this is great, thank you! I just linked your video from the main page an 
from the wiki. I also used a screenshot out of your video for the link. 
But github may need some time to render this.

Great job!

**Reply by deadesq** [reply]
Thanks!  So...my partner thought all this was really interesting, and she wondered who the people are who are working on this project.  I told her your user name, but she thought it odd that this is all I know about you--anyway, I'm Dave, in California, and I'm not really a programmer.  Just a professional photographer with a need for a solution.

**Comment by outdoorbits** [comment]
The browser is defined in 
/home/lbb-desktop/.config/autostart/little-backup-box.desktop

If you edit it by

sudo nano /home/lbb-desktop/.config/autostart/little-backup-box.desktop

you can change firefox to whatever you like. I could set 
chromium-browser to start on boot this way:

Exec=chromium-browser http://localhost

But it fails when using --start-fullscreen or --kiosk.

I think, the necessary command would be

Exec=chromium-browser http://localhost --start-fullscreen --kiosk 
--noerrdialogs --no-first-run

Executing this command from terminal, I got some error messages, so I 
had to install upower:

sudo apt install upower

This fixed some of these errors. But it still didn't work - and the 
error messages about missing objects where not helpful to me anymore.

That was why I decided to use firefox. Maybe you find the error?

**Comment by outdoorbits** [comment]
That's nice! The idea of little-backup-box was born by Dmitri Popov, 
maybe about 10 years ago. The scripts where written in bash script 
language and he already had a web ui and a display working. Dmitri even 
published in magazines about lbb. When I found this project it seamed to 
be just what I have been looking for since a long time. It was really 
great. But using it I quickly missed some functions, flexibility, some 
security checks, cloud functions and ... and started to contribute. One 
day Dmitri asked me to take over the project... Now it is nearly 
completely rewritten. The bash scripts are mostly replaced by python but 
the look and feel is still close to Dmitris work.

I'm Stefan and live in Germany. I'm just a hobby photographer. Most 
images we take on traveling - and that's why I need a backup solution. 
Usually we travel as family and carry cameras and smartphones. So I 
enjoy just to plug all these devices into my USB hub in the evening and 
if the internet connection is well, all the images are save at home next 
morning.

But you are a professional photographer, in California? Not bad!!! 
Sounds extremely interesting! Do you have a website for presenting your 
images?

**Reply by outdoorbits** [reply]
While watching your video I thought how easy it was not to use the touchscreen but a tiny mouse with scroll wheel. Did you ever think about that?

**Reply by deadesq** [reply]
Actually.....one step further.  I keep this in the little kit I've assembled to travel with this.  It's helpful to navigate out of the full screen kiosk mode when the display freezes for whatever reason.
![IMG_1660](https://github.com/outdoorbits/little-backup-box/assets/153684931/580dbcba-2fb2-4ff5-b3c5-5752ceed7c4e)

**Reply by deadesq** [reply]
And I forgot to reply to your question about my website--it's at https://justalittlelight.com

**Reply by outdoorbits** [reply]
I would like to create a kind of gallery. Could I use this image?

**Reply by deadesq** [reply]
Sure--glad to help.

**Comment by outdoorbits** [comment]
Great photographs! On images like them I would love to know, what effort it took to get them, for sure a lot!

**Reply by deadesq** [reply]
Thank you!  Some of the images are easier to get than others.  I do have to travel quite far for the wolves, polar bears and brown bears, but the black bears are in my back yard all summer, so those are easy.

**Comment by outdoorbits** [comment]
Thank you, it's published right now: 
https://github.com/outdoorbits/little-backup-box/blob/main/gallery.md
